r-package/grf/tests/testthat/test_causal_forest.R

```diff
@@ -125,12 +125,56 @@ test_that("local linear causal forests with large lambda are equivalent to causa
    p = 6
    X = matrix(rnorm(n*p), n, p)
    W = rbinom(n, 1, 0.5)
-   TAU = 2*X[,1] + X[,2]
-   Y = W * TAU + rnorm(n)
+   tau= 2*X[,1] + X[,2]
+   Y = W * tau + rnorm(n)
 
    forest = causal_forest(X, Y, W, num.trees = 400)
    preds.ll = predict(forest, X, linear.correction.variables = 1:ncol(X), ll.lambda = 1e5)$predictions
    preds.cf = predict(forest)$predictions
 
    expect_true(mean((preds.ll - preds.cf)^2) < 0.02)
 })
+
+
+test_that("predictions are invariant to scaling of the sample weights.", {
+   n = 1000
+   p = 6
+   X = matrix(rnorm(n*p), n, p)
+   e = 1/(1+exp(-3*X[,1]+3*X[,2]))
+   W = rbinom(n, 1, e)
+   tau = 2*X[,1] + X[,2]
+   Y = W * tau + rnorm(n)
+   e.cc = 1/(1+exp(-3*X[,1]))
+   sample.weights = 1/e.cc
+
+   forest.1 = causal_forest(X, Y, W, sample.weights = sample.weights)
```

Comment:
At least some of these sample sizes are roughly calibrated so that we get a meaningful test. When sample sizes are much smaller, I have to increase overlap between the censored and uncensored populations, so that the ones dealing with censoring aren't so variable that the tests are meaningless. At some point, this makes the bias due to censoring small enough that the weights don't have much to correct for. I'll see what I can do. Definitely fewer trees.